service/jmx/src/main/java/org/jolokia/service/jmx/handler/WriteHandler.java

```diff
@@ -132,21 +133,15 @@ private Object setAttribute(JolokiaWriteRequest request, MBeanServerConnection s
 
             // We set an inner value, hence we have to return provided value itself for resetting
             // it later back via JMX
-            return new Object[] {
-                    pCurrentValue,
-                    context.getMandatoryService(Serializer.class).setInnerValue(pCurrentValue, newValue, pathParts)
-            };
+            return context.getMandatoryService(Serializer.class).setInnerValue(pCurrentValue, newValue, pathParts);
```

Comment:
Hum, i m not especially attached to typing things here. I ll try to explain the problem I attempted to solve so maybe you ll have better ideas. 

When implementing the gson/jsonpath based serializer I was confused about how the setInnerValue was working since only the old value was returned. It took me some time to figure out that the return value does not have a lot of importance and the method in `JolokiaSerializer` is basically mutating the `pOuterObject` and then the caller sets it properly on a bean. 

This approach is valid but still it makes quite complex (and long) to implement a custom serializer without reimplementing all the reflexion mechanics JolokiaSerializer has to find out the right operation and so on. 

This is why I tried to give setInnerValue a bit more function-like behavior taking some read-only parameters and returning a value (a pair of values actually). This approach makes things easier for the serializer implementor.

If you do prefer keeping the "untyped" approach we could easily replace WriteRequestValues by an `Object[]`. 

Please let me know what do you think.